Server/mods/deathmatch/logic/CBuildingManager.h

```diff
@@ -0,0 +1,45 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/deathmatch/logic/CBuildingManager.h
+ *  PURPOSE:     Building entity manager class
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#pragma once
+
+#include "CBuilding.h"
+#include <list>
+
+using std::list;
+
+class CBuildingManager
+{
+    friend class CBuilding;
+public:
+    CBuildingManager();
+    ~CBuildingManager();
+
+    CBuilding* Create(CElement* pParent);
+    CBuilding* CreateFromXML(CElement* pParent, CXMLNode& Node, CEvents* pEvents);
+    void     DeleteAll();
+
+    unsigned int Count() { return static_cast<unsigned int>(m_List.size()); };
+    bool         Exists(CBuilding* pObject) const;
+
+    using CBuildingListType = CFastList<CBuilding*>;
+    CBuildingListType::const_iterator IterBegin() const { return m_List.begin(); };
+    CBuildingListType::const_iterator IterEnd() const { return m_List.end(); };
```

Comment:
It should share the same interface with another managers. It's useful for lambda functions

Reply:
> It should share the same interface with another managers.

Understandable. Make at least a begin and end methods for standard range based loop

> It's useful for lambda functions

Not really since for range loops exist